### PR TITLE
[SPARK-30908][CORE][DOC] Add version information to the configuration of Kryo

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
1.Add version information to the configuration of `Kryo`.
2.Update the docs of `Kryo`.

I sorted out some information show below.

Item name | Since version | JIRA ID | Commit ID | Note
-- | -- | -- | -- | --
spark.kryo.registrationRequired | 1.1.0 | SPARK-2102 | efdaeb111917dd0314f1d00ee8524bed1e2e21ca#diff-1f81c62dad0e2dfc387a974bb08c497c |  
spark.kryo.registrator | 0.5.0 | None | 91c07a33d90ab0357e8713507134ecef5c14e28a#diff-792ed56b3398163fa14e8578549d0d98 | This is not a release version, do we need to record it?
spark.kryo.classesToRegister | 1.2.0 | SPARK-1813 | 6bb56faea8d238ea22c2de33db93b1b39f492b3a#diff-529fc5c06b9731c1fbda6f3db60b16aa |  
spark.kryo.unsafe | 2.1.0 | SPARK-928 | bc167a2a53f5a795d089e8a884569b1b3e2cd439#diff-1f81c62dad0e2dfc387a974bb08c497c |  
spark.kryo.pool | 3.0.0 | SPARK-26466 | 38f030725c561979ca98b2a6cc7ca6c02a1f80ed#diff-a3c6b992784f9abeb9f3047d3dcf3ed9 |  
spark.kryo.referenceTracking | 0.8.0 | None | 0a8cc309211c62f8824d76618705c817edcf2424#diff-1f81c62dad0e2dfc387a974bb08c497c |  
spark.kryoserializer.buffer | 1.4.0 | SPARK-5932 | 2d222fb39dd978e5a33cde6ceb59307cbdf7b171#diff-1f81c62dad0e2dfc387a974bb08c497c |  
spark.kryoserializer.buffer.max | 1.4.0 | SPARK-5932 | 2d222fb39dd978e5a33cde6ceb59307cbdf7b171#diff-1f81c62dad0e2dfc387a974bb08c497c |  

### Why are the changes needed?
Supplemental configuration version information.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Exists UT

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/internal/config/Kryo.scala [hunk 1]
```diff
@@ -22,35 +22,43 @@ import org.apache.spark.network.util.ByteUnit
 private[spark] object Kryo {
 
   val KRYO_REGISTRATION_REQUIRED = ConfigBuilder("spark.kryo.registrationRequired")
+    .version("1.1.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Kryo.scala]: SPARK-2102, commit ID: efdaeb111917dd0314f1d00ee8524bed1e2e21ca#diff-1f81c62dad0e2dfc387a974bb08c497c

File: core/src/main/scala/org/apache/spark/internal/config/Kryo.scala [hunk 2]
```diff
@@ -22,35 +22,43 @@ import org.apache.spark.network.util.ByteUnit
...
     .booleanConf
     .createWithDefault(false)
 
   val KRYO_USER_REGISTRATORS = ConfigBuilder("spark.kryo.registrator")
+    .version("0.5.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Kryo.scala]: No JIRA ID, commit ID: 91c07a33d90ab0357e8713507134ecef5c14e28a#diff-792ed56b3398163fa14e8578549d0d98
This is not a release version, do we need to record it?

File: core/src/main/scala/org/apache/spark/internal/config/Kryo.scala [hunk 3]
```diff
@@ -22,35 +22,43 @@ import org.apache.spark.network.util.ByteUnit
...
     .stringConf
     .createOptional
 
   val KRYO_CLASSES_TO_REGISTER = ConfigBuilder("spark.kryo.classesToRegister")
+    .version("1.2.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Kryo.scala]: SPARK-1813, commit ID: 6bb56faea8d238ea22c2de33db93b1b39f492b3a#diff-529fc5c06b9731c1fbda6f3db60b16aa

File: core/src/main/scala/org/apache/spark/internal/config/Kryo.scala [hunk 4]
```diff
@@ -22,35 +22,43 @@ import org.apache.spark.network.util.ByteUnit
...
     .stringConf
     .toSequence
     .createWithDefault(Nil)
 
   val KRYO_USE_UNSAFE = ConfigBuilder("spark.kryo.unsafe")
+    .version("2.1.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Kryo.scala]: SPARK-928, commit ID: bc167a2a53f5a795d089e8a884569b1b3e2cd439#diff-1f81c62dad0e2dfc387a974bb08c497c

File: core/src/main/scala/org/apache/spark/internal/config/Kryo.scala [hunk 5]
```diff
@@ -22,35 +22,43 @@ import org.apache.spark.network.util.ByteUnit
...
     .booleanConf
     .createWithDefault(false)
 
   val KRYO_USE_POOL = ConfigBuilder("spark.kryo.pool")
+    .version("3.0.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Kryo.scala]: SPARK-26466, commit ID: 38f030725c561979ca98b2a6cc7ca6c02a1f80ed#diff-a3c6b992784f9abeb9f3047d3dcf3ed9

File: core/src/main/scala/org/apache/spark/internal/config/Kryo.scala [hunk 6]
```diff
@@ -22,35 +22,43 @@ import org.apache.spark.network.util.ByteUnit
...
     .booleanConf
     .createWithDefault(true)
 
   val KRYO_REFERENCE_TRACKING = ConfigBuilder("spark.kryo.referenceTracking")
+    .version("0.8.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Kryo.scala]: No JIRA ID, commit ID: 0a8cc309211c62f8824d76618705c817edcf2424#diff-1f81c62dad0e2dfc387a974bb08c497c

File: core/src/main/scala/org/apache/spark/internal/config/Kryo.scala [hunk 7]
```diff
@@ -22,35 +22,43 @@ import org.apache.spark.network.util.ByteUnit
...
     .booleanConf
     .createWithDefault(true)
 
   val KRYO_SERIALIZER_BUFFER_SIZE = ConfigBuilder("spark.kryoserializer.buffer")
+    .version("1.4.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Kryo.scala]: SPARK-5932, commit ID: 2d222fb39dd978e5a33cde6ceb59307cbdf7b171#diff-1f81c62dad0e2dfc387a974bb08c497c

File: core/src/main/scala/org/apache/spark/internal/config/Kryo.scala [hunk 8]
```diff
@@ -22,35 +22,43 @@ import org.apache.spark.network.util.ByteUnit
...
     .bytesConf(ByteUnit.KiB)
     .createWithDefaultString("64k")
 
   val KRYO_SERIALIZER_MAX_BUFFER_SIZE = ConfigBuilder("spark.kryoserializer.buffer.max")
+    .version("1.4.0")
```
Comment (beliefer) [inline core/src/main/scala/org/apache/spark/internal/config/Kryo.scala]: SPARK-5932, commit ID: 2d222fb39dd978e5a33cde6ceb59307cbdf7b171#diff-1f81c62dad0e2dfc387a974bb08c497c

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by beliefer** [comment]
@HyukjinKwon Thanks for your work.

**Comment by HyukjinKwon** [comment]
Merged to branch-3.0 too

**Pull request closed without merging**